libs/Texture/Texture.cpp

```diff
@@ -24,68 +23,128 @@ GLuint Texture::GetTextureId()
 	return m_texture_id;
 }
 
+void Texture::UnBind()
+{
+	glActiveTexture(GL_TEXTURE0);
+	glBindTexture(m_texture_target, 0);
+}
+
 int Texture::Load2DTexture()
 {
-	
 
 	int width = 0;
 	int height = 0;
 	int bbp = 0;
-	//pointer to the image data
-	unsigned char * bits(0);
+	// pointer to the image data
+	unsigned char *bits(0);
 
-	//retrieve the image data
-	//if this somehow one of these failed (they shouldn't), return failure
+	// retrieve the image data
+	// if this somehow one of these failed (they shouldn't), return failure
 	GLuint textureID = LoadTexture(m_file_name, width, height, &bits, bbp);
 	if (textureID == -1)
 	{
 		assert(false && "Image failed to load 2");
 	}
 
-	
 	return textureID;
 }
 
-int Texture::LoadTexture(const std::string & fileName, 
-	int & width, int & height, unsigned char ** data, int& bbp)
+int Texture::Load3DTexture(const std::vector<std::string> &paths)
+{
+	if (paths.size() == 0)
+	{
+		assert(false && "no paths to load from");
```

Comment:
why the hardcoded false here?

Reply:
it's a trick to get the message printed by the `assert`